package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,414 @@
+            diffusion map. More information in [1], [2] and [3].
+        weights: list, optional
+            The list has to have the same length as the trajectory.
+            With 'None' the weight of each frame of the trajectory will be the
+            same.
+        timescale: int, optional
+            The number of steps in the random walk, large t reflects global
+            structure whereas small t indicates local structure.
+        """
+        self.DistanceMatrix = DistanceMatrix
+        self._nframes = DistanceMatrix.nframes
+        self._t = timescale
+        self._epsilon = epsilon
+
+        if weights is None:
+            # weights do not apply to metric but density of data
+            self._weights_ker = np.ones((self._nframes,))
+        else:
+            if weights.shape[0] != self._nframes:
+                raise ValueError("The weight should have the same length as "
+                                 'the trajectory')
+            else:
+                # weights are constructed as relative to the mean
+                self._weights_ker = (np.asarray(weights, dtype=np.float64) /
+                                     np.mean(weights))
+
+    def decompose_kernel(self):
+        # this should be a reference to the same object as
+        # self.DistanceMatrix.dist_matrix
+        self._kernel = self._epsilon.scaledMatrix
+
+        # take negative exponent of scaled matrix to create Isotropic kernel
+        self._kernel = np.exp(-self._kernel)
+
+        # define an anistropic diffusion term q
+        q_vector = np.zeros((self._nframes, ))
+        # weights should reflect the density of the points on the manifold
+        for i in range(self._nframes):
+            q_vector[i] = np.dot(self._kernel[i, :], self._weights_ker)
+
+        # Form a new kernel from the anisotropic diffusion term q
+        self._kernel /= np.sqrt(q_vector[:, np.newaxis].dot(q_vector[np.newaxis]))
+
+        # Weighted Graph Laplacian normalization on this new graph
+        d_vector = np.zeros((self._nframes, ))
+        for i in range(self._nframes):
+            d_vector[i] = np.dot(self._kernel[i, :], self._weights_ker)
+
+        for i in range(self._nframes):
+            self._kernel[i, :] = self._kernel[i, :] * self._weights_ker
+
+        # Define anisotropic transition by dividing kernel by this term
+        self._kernel /= np.sqrt(d_vector[:, np.newaxis].dot(d_vector[np.newaxis]))
+
+        # Apply timescaling
+        for i in range(self._t):
+            if i > 0:
+                self._kernel = self._kernel.dot(self._kernel.T)
+
+        eigenvals, eigenvectors = np.linalg.eig(self._kernel)
+
+        eg_arg = np.argsort(eigenvals)
+        self.eigenvalues = eigenvals[eg_arg[::-1]]
+        self.eigenvectors = eigenvectors[eg_arg[::-1], :]
+
+    def _prepare(self):
+        self.fit = np.zeros((self.eigenvectors.shape[0],
+                             self.num_eigenvectors))
+
+    def _single_frame(self):
```

Comment:
OK it does. But this doesn't make any sense here. Just calling the `run` method without calling decompose now would just lead to some garbage. Then there is the issue that this would lead to do a transformation for all coordinates when run is called. There is no way to decide on the dimension used.